fdbclient/NativeAPI.actor.cpp

```diff
@@ -3124,38 +3129,39 @@ ACTOR Future<Void> warmRange_impl(Reference<TransactionState> trState, KeyRange
 	return Void();
 }
 
-SpanID generateSpanID(bool transactionTracingSample, SpanID parentContext = SpanID()) {
-	uint64_t txnId = deterministicRandom()->randomUInt64();
+// TODO - ljoswiak and mpilman. This will need some close inspection.
+SpanContext generateSpanID(bool transactionTracingSample, SpanContext parentContext = SpanContext()) {
+	UID txnId = UID(deterministicRandom()->randomUInt64(), deterministicRandom()->randomUInt64());
 	if (parentContext.isValid()) {
-		if (parentContext.first() > 0) {
-			txnId = parentContext.first();
+		if (parentContext.traceID.first() > 0) {
```

Comment:
I think this conditional is redundant now. If the above condition `parentContext.isValid()` is true, then `parentContext.traceID.first()` will always be nonzero.